index.d.ts

```diff
@@ -0,0 +1,81 @@
+export interface Action {
+  type: string;
+}
+
+
+/* reducers */
+
+export type Reducer<S> = <A extends Action>(state: S, action: A) => S;
+
+export function combineReducers<S>(reducers: {[key: string]: Reducer<any>}): Reducer<S>;
+
+
+/* store */
+
+export interface Dispatch {
+  <A>(action: A): A;
+  <A, B>(action: A): B;
+}
+
+export interface Store<S> {
+  dispatch: Dispatch;
+  getState(): S;
+  subscribe(listener: () => void): () => void;
```

Comment:
I did not test it but i think that `Function` has few things like `bind` that are not declared with just call signature… But I may be wrong. It may be redundant anyway.